src/AccentColor/AccentColorManager.vala

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright 2021 elementary, Inc. (https://elementary.io)
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA
+ *
+ * Authored by: Marius Meisenzahl <mariusmeisenzahl@gmail.com>
+ */
+
+ public class Gala.AccentColorManager : Object {
+    private const string INTERFACE_SCHEMA = "org.gnome.desktop.interface";
+    private const string STYLESHEET_KEY = "gtk-theme";
+    private const string STYLESHEET_PREFIX = "io.elementary.stylesheet.";
+    private const string TAG_ACCENT_COLOR = "Xmp.xmp.io.elementary.AccentColor";
+
+    private Gala.AccountsService? gala_accounts_service = null;
+
+    private Settings background_settings;
+    private Settings interface_settings;
+
+    private NamedColor[] theme_colors = {
```

Comment:
Instead of hardcoding the theme colors here, you could use the foreign drawing API here to get the theme color. Have a look to `InternalUtils.get_theme_accent_color`. You could change it a little bit so it allows to select a theme:

```csharp
        private static Gtk.StyleContext selection_style_context = null;
        public static Gdk.RGBA get_theme_accent_color (string? theme_name = null) {
            if (selection_style_context == null) {
                var label_widget_path = new Gtk.WidgetPath ();
                label_widget_path.append_type (GLib.Type.from_name ("label"));
                label_widget_path.iter_set_object_name (-1, "selection");

                selection_style_context = new Gtk.StyleContext ();

                if (theme_name != null) {
                    unowned Gtk.CssProvider theme_provider = Gtk.CssProvider.get_named (theme_name, null);
                    selection_style_context.add_provider (theme_provider, Gtk.STYLE_PROVIDER_PRIORITY_USER);
                }

                selection_style_context.set_path (label_widget_path);
            }

            return (Gdk.RGBA) selection_style_context.get_property (
                Gtk.STYLE_PROPERTY_BACKGROUND_COLOR,
                Gtk.StateFlags.NORMAL
            );
        }
```

Then, you could create a enum of themes with values like `"io.elementary.stylesheet.mint"` and use those as the  `theme_name` argument.

Watch out that `selection_style_context` caching, not sure why it is there. You may need to create a different method.

Reply:
Thanks for the tip! I have created a new method based on your suggestion. Works flawlessly 🤓️